src/main/java/net/minecraftforge/client/model/ModelDataManager.java

```diff
@@ -0,0 +1,101 @@
+package net.minecraftforge.client.model;
+
+import java.lang.ref.WeakReference;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.base.Preconditions;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.data.IModelData;
+
+public class ModelDataManager
+{
+    private static WeakReference<World> currentWorld = new WeakReference<>(null);
```

Comment:
Can just do explicit `WeakReference<WorldClient>`

Reply:
There's no need, this would just add a classloading dependency on a server-unsafe class for no reason. I don't use any methods of WorldClient.